sceptre/template_handlers/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+import abc
+import logging
+
+import six
+from jsonschema import validate, ValidationError
+
+from sceptre.exceptions import TemplateHandlerArgumentsInvalidError
+
+
+@six.add_metaclass(abc.ABCMeta)
+class TemplateHandler:
+    """
+    TemplateHandler is an abstract base class that should be inherited
+    by all Template Handlers.
+
+    :param name: Name of the template
+    :type name: str
+
+    :param arguments: The arguments of the template handler
+    :type arguments: dict
+
+    :param sceptre_user_data: Sceptre user data in stack config
+    :type sceptre_user_data: dict
+
+    :param connection_manager: Connection manager used to call AWS
+    :type connection_manager: sceptre.connection_manager.ConnectionManager
+    """
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, name, arguments=None, sceptre_user_data=None, connection_manager=None):
+        self.logger = logging.getLogger(__name__)
+        self.name = name
+        self.arguments = arguments
+        self.sceptre_user_data = sceptre_user_data
+        self.connection_manager = connection_manager
```

Comment:
I feel like I'd really want the whole stack (like a hook has) if I were implementing a template handler. Things like the S3 bucket name and the template key prefix, region, etc... I suppose those could be sent to the user data... But that's a bit awkward. For a template handler that could do something like a Sam package, I'd want all that stuff.

Reply:
This PR is quite large already @jfalkenstein, lets do that in a separate PR?